### PR TITLE
Fix #13

### LINKED ISSUE #13: `OnBuildError` is not called if `PrepareBuild` throws an exception
`OnBuildError` *is* called if [`BuildPipeline.BuildPlayer`](https://docs.unity3d.com/ScriptReference/BuildPipeline.BuildPlayer.html) (called within [`BuildSync`](https://github.com/sttz/trimmer/blob/master/Editor/BuildManager.cs#L450)) fails. The same cannot be said if `PrepareBuild` fails.

### PR DESCRIPTION
**Pull request by JesseTG**
Here, what do you think about this?

- Call OnBuildError if PrepareBuild or GetScriptingDefineSymbols throws an exception
- Also call it if the user didn't (or couldn't) pick a build location if it wasn't already given
- Mark the argument to OnBuildError as [CanBeNull]

### CODE REVIEW

File: Editor/BuildManager.cs [hunk 1]
```diff
@@ -457,25 +458,35 @@ public static BuildReport BuildSync(BuildProfile buildProfile, BuildPlayerOption
...
 
-        // Add Trimmer scripting define symbols
-        AddScriptingDefineSymbols(buildProfile, ref options);
-
-        // Run options' PrepareBuild
-        foreach (var option in GetCurrentEditProfile().OrderBy(o => o.PostprocessOrder)) {
-            if ((option.Capabilities & OptionCapabilities.ConfiguresBuild) == 0) continue;
-            var inclusion = buildProfile == null ? OptionInclusion.Remove : buildProfile.GetInclusionOf(option, options.target);
-            options = option.PrepareBuild(options, inclusion);
+        try
```
Comment (sttz) [inline Editor/BuildManager.cs]: Can you change the try/catch formatting here to follow the K&R style of the rest of Trimmer? (i.e. don't put the curly braces on separate lines.)

Comment (JesseTG) [reply Editor/BuildManager.cs]: Oops, I must have missed that. On it.

**Comment by sttz** [comment]
Great, thanks for the fix!

**Comment by JesseTG** [comment]
No problem, thanks for Trimmer.

**Pull request merged**